apps/finance/app/src/components/TransferRow.js

```diff
@@ -48,136 +48,110 @@ class TransferRow extends React.Component {
+
     const formattedAmount = formatTokenAmount(
       amount,
       isIncoming,
-      decimals,
+      token.decimals,
       true,
       { rounding: 5 }
     )
     const formattedDate = format(date, "yyyy-MM-dd'T'HH:mm:ss.SSSxxx")
+
     return (
-      <TableRow>
-        <NoWrapCell>
-          <time dateTime={formattedDate} title={formattedDate}>
-            {format(date, 'dd/MM/yy')}
-          </time>
-        </NoWrapCell>
-        <NoWrapCell>
-          <IdentityBadge
-            networkType={network.type}
-            entity={entity}
-            shorten={!wideMode}
-          />
-        </NoWrapCell>
-        <NoWrapCell title={reference} style={{ position: 'relative' }}>
-          <TextOverflow style={{ position: 'absolute', left: '0', right: '0' }}>
-            {reference}
-          </TextOverflow>
-        </NoWrapCell>
-        <NoWrapCell align="right">
-          <Amount positive={isIncoming}>
-            {formattedAmount} {symbol}
-          </Amount>
-        </NoWrapCell>
-        <NoWrapCell>
-          <ActionsWrapper>
-            {etherscanBaseUrl && (
-              <ContextMenu>
-                {/* <ContextMenuItem onClick={this.handleCopyTransferUrl}>
-                <IconShare />
-                <ActionLabel>Copy Transfer URL</ActionLabel>
-              </ContextMenuItem> */}
-                <ContextMenuItem onClick={this.handleViewTransaction}>
-                  <IconTokens />
-                  <ActionLabel>View Transaction</ActionLabel>
-                </ContextMenuItem>
-              </ContextMenu>
-            )}
-            {showCopyTransferMessage && (
-              <ConfirmMessageWrapper>
-                <ConfirmMessage onDone={this.handleConfirmMessageDone}>
-                  Transaction URL copied to clipboard
-                </ConfirmMessage>
-              </ConfirmMessageWrapper>
-            )}
-          </ActionsWrapper>
-        </NoWrapCell>
-      </TableRow>
+      <Viewport>
+        {({ below }) =>
+          below('medium') ? (
+            <TableRow>
+              <StyledTableCell>
+                <Grid>
+                  <div>
+                    <div css="display: flex">
+                      <IdentityBadge
+                        networkType={network.type}
+                        entity={entity}
+                        shorten
+                      />
+                    </div>
+                  </div>
+                  <time dateTime={formattedDate} title={formattedDate}>
+                    {format(date, 'dd MMM yyyy')}
+                  </time>
+                  <TextOverflow style={{ marginTop: '5px' }}>
+                    {reference}
+                  </TextOverflow>
+                  <Amount positive={isIncoming} style={{ marginTop: '5px' }}>
+                    {formattedAmount} {token.symbol}
+                  </Amount>
+                </Grid>
+              </StyledTableCell>
+            </TableRow>
+          ) : (
```

Comment:
I would say in this case it might make sense to have explicit `if/else` blocks, since the tree is so large and it's hard to spot the `:`